apps/remix-ide/src/app/files/workspaceFileProvider.js

```diff
@@ -33,7 +33,7 @@ class WorkspaceFileProvider extends FileProvider {
     if (!this.workspace) this.createWorkspace()
     path = path.replace(/^\/|\/$/g, '') // remove first and last slash
     if (path.startsWith(this.workspacesPath + '/' + this.workspace)) return path
-    if (path.startsWith(this.workspace)) return this.workspacesPath + '/' + this.workspace
+    if (path.startsWith(this.workspace)) return path.replace(this.workspace, this.workspacesPath + '/' + this.workspace)
```

Comment:
i don't understand this change, is this a fix for when a folder has the same name as the workspace?

Reply:
This fixes an issue i had when querying provider.exists(). If i have a path like `default_workspace/contracts/ballot.sol`, before resolving if the path exists `removePrefix()` is called. Now with the previous check, since the path starts with `default_workspace`, `.workspaces/default_workspace` is returned and when this (`.workspaces/default_workspace`) is checked if it exists, it always returns **true.** My change fixes that.